src/ray/core_worker/core_worker.cc

```diff
@@ -1982,13 +2001,9 @@ std::vector<rpc::ObjectReference> CoreWorker::SubmitActorTask(
   } else {
     returned_refs = task_manager_->AddPendingTask(
         rpc_address_, task_spec, CurrentCallSite(), actor_handle->MaxTaskRetries());
-    io_service_.post(
-        [this, task_spec]() {
-          RAY_UNUSED(direct_actor_submitter_->SubmitTask(task_spec));
-        },
-        "CoreWorker.SubmitActorTask");
+    direct_actor_submitter_->AppendTask(task_spec);
```

Comment:
any reason why do we change the threading model here?

Reply:
As mentioned in the [design doc](https://docs.google.com/document/d/1OGkiOJ-mIxmgNNigO9E7XTCZxzXaj-36u1hQQnAjZgg/edit#), We made a few changes to the threading model:
> Adjust the sending thread model, if the preconfigured max length threshold is reached, the user thread returns an error immediately to indicate the remote call is failed due to backpressure. **If not, the user thread puts TaskSpec object into client queue and then post sending lambda to IO thread, and sent out by IO thread**. 

`AppendTask` does the first half, put task spec into client queue and do post, while `SubmitTask`does the second half, send task spec out, executing in `actor_task_io_service_`. Indeed, `actor_task_io_service_` should not be introduced, we can reuse the io_service_ in CoreWorker. In this way, we can reduce the modification of the threading model, no new thread introduced.

Reply:
>  any reason why we change the threading model here?



This PR doesn't change the threading model, we just move some unnecessary code to sync codepath from async codepath in this PR.

Before this PR, we post `direct_actor_transport->SubmiTask()` into ioservice, but after this PR, we need to throw exceptions to user when they invoking `f.remote()` synchronously. So we need move this post https://github.com/ray-project/ray/blob/master/src/ray/core_worker/core_worker.cc#L1994 to https://github.com/ray-project/ray/blob/master/src/ray/core_worker/transport/direct_actor_transport.cc#L87